posthog/session_recordings/queries/session_recording_list_from_filters.py

```diff
@@ -245,17 +228,121 @@ def _having_predicates(self) -> ast.And | Constant:
+                event_names.update([ae for ae in action.get_step_events() if ae not in event_names])
+            else:
+                if entity.id and entity.id not in event_names:
+                    event_names.add(entity.id)
+
+            # TODO: we're not passing the "right" type in here - should we change the signature or do something else?
+            entity_exprs = [entity_to_expr(entity=entity)]  # type: ignore
+
+            if entity.property_groups:
+                entity_exprs.append(property_to_expr(entity.property_groups, team=self._team, scope="replay"))
+
+            event_exprs.append(ast.And(exprs=entity_exprs))
+
+        return event_exprs, list(event_names)
+
+    def get_query(self):
+        return ast.SelectQuery(
+            select=[ast.Alias(alias="session_id", expr=ast.Field(chain=["$session_id"]))],
+            select_from=ast.JoinExpr(table=ast.Field(chain=["events"])),
+            prewhere=self._prewhere_predicates(),
+            where=self._where_predicates(),
+            having=self._having_predicates(),
+            group_by=[ast.Field(chain=["$session_id"])],
+        )
+
+    def _prewhere_predicates(self) -> ast.Expr:
+        exprs: list[ast.Expr] = [
+            # regardless of any other filters limit between TTL and current time
+            ast.CompareOperation(
+                op=ast.CompareOperationOp.GtEq,
+                left=ast.Field(chain=["timestamp"]),
+                right=ast.Constant(value=datetime.now() - timedelta(days=self._ttl_days)),
+            ),
+            ast.CompareOperation(
+                op=ast.CompareOperationOp.LtEq,
+                left=ast.Field(chain=["timestamp"]),
+                right=ast.Call(name="now", args=[]),
+            ),
+        ]
+
+        if self._filter.date_from:
             exprs.append(
-                ast.Call(
-                    name="hasAll",
-                    args=[
-                        ast.Call(name="groupUniqArray", args=[ast.Field(chain=["events", "event"])]),
-                        # KLUDGE: sorting only so that snapshot tests are consistent
-                        ast.Constant(value=sorted(event_names)),
-                    ],
+                ast.CompareOperation(
+                    op=ast.CompareOperationOp.GtEq,
+                    left=ast.Field(chain=["timestamp"]),
+                    right=ast.Constant(value=self._filter.date_from - timedelta(hours=12)),
                 )
             )
 
-        return ast.And(exprs=exprs) if exprs else Constant(value=True)
+        if self._filter.date_from:
+            exprs.append(
+                ast.CompareOperation(
+                    op=ast.CompareOperationOp.LtEq,
+                    left=ast.Field(chain=["timestamp"]),
+                    right=ast.Constant(value=self._filter.date_to + timedelta(hours=12)),
+                )
+            )
+
+        return ast.And(exprs=exprs)
+
+    def _where_predicates(self) -> ast.Expr:
```

Comment:
I specifically left the [`persons_sub_query`](https://github.com/PostHog/posthog/blob/master/posthog/session_recordings/queries/session_recording_list_from_replay_summary.py#L548-L564) out of this where clause. It seemed related to PoE (based on the underlying [ActorsQuery](https://github.com/PostHog/posthog/blob/5fe209535efc9758b0414c12e8c9cafc3e0c9d56/posthog/session_recordings/queries/session_recording_list_from_replay_summary.py#L160-L260)).

I didn't have full context to know whether it was needed (maybe @pauldambra you know why it was added) or perhaps HogQL takes care of things under the hood

Reply:
_hopefully_ it's covered by HogQL

I know there were some customers that couldn't filter on person properties without PoE working but we can make sure we test with them before rolling out the flag and get some test cases here to prove the queries (if they don't exist already)